data_structures/linked_list/detecting_cycle_in_list.c

```diff
@@ -0,0 +1,51 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct Node {
+	int data;
+	struct Node* next;
+};
+
+void push(struct Node** head_ref, int new_data)
+{
+	struct Node* new_node
+		= (struct Node*)malloc(sizeof(struct Node));
+
+	new_node->data = new_data;
+
+	new_node->next = (*head_ref);
+
+	(*head_ref) = new_node;
+}
+
+int detectLoop(struct Node* list)
+{
+	struct Node *slow_p = list, *fast_p = list;

```

Comment:
If I remember correctly, It's a Floyd's tortoise and hare algorithm. Probably it's better to add a comment about this. That would be easy to recognize it in the future.